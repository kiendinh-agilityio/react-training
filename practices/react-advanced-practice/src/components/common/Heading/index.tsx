import { Heading as HeadingBase, HeadingProps } from '@radix-ui/themes';
import { memo } from 'react';

// import types
import { TextSize } from '@/types';

type HeadingBaseProps = {
  text: string;
  size?: TextSize;
} & HeadingProps;

const Heading = memo(
  ({
    text,
    as = 'h1',
    weight = 'bold',
    size = TextSize.Large,
    className = '',
  }: HeadingBaseProps) => (
    <HeadingBase
      as={as}
      weight={weight}
      className={`text-${size} font-bold text-dark ${className}`}
    >
      {text}
    </HeadingBase>
  ),
);

export default Heading;
