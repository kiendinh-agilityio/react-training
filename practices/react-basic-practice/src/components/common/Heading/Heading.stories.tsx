import Heading from '.';

export default {
  component: Heading,
  title: 'Heading',
  tags: ['autodocs'],
};

export const Tables = {
  args: {
    text: 'Authors Table',
  },
};
