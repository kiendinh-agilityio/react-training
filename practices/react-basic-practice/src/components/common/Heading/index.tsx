interface HeadingProps {
  text: string;
  level?: 1 | 2 | 3;
  size: 'sm' | 'lg' | 'xl';
  className?: string;
}

const Heading = ({ text, level = 1, className = '', size }: HeadingProps) => {
  const Tag = `h${level}` as keyof JSX.IntrinsicElements;

  return (
    <Tag className={`text-${size} flex justify-center font-helveticaBold font-bold ${className}`}>
      {text}
    </Tag>
  );
};

export default Heading;
