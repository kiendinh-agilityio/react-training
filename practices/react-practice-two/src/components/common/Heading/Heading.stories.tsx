import { StoryFn, Meta } from '@storybook/react';
import Heading from '.';

export default {
  title: 'Heading',
  component: Heading,
  argTypes: {
    level: {
      control: { type: 'select', options: [1, 2, 3] },
    },
    spanContent: {
      control: 'text',
    },
  },
  parameters: {
    layout: 'centered',
  },
} as Meta;

type HeadingStoryProps = {
  children: React.ReactNode;
  level: 1 | 2 | 3;
};

const Template: StoryFn<HeadingStoryProps> = (args: HeadingStoryProps) => <Heading {...args} />;

// H1 Story
export const H1 = Template.bind({});
H1.args = {
  level: 1,
  children: 'THE FUTURE.',
};

// H2 Story
export const H2 = Template.bind({});
H2.args = {
  children: 'Product Information',
  level: 2,
};
