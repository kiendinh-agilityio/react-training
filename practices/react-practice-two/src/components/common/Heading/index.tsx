import { TextSize } from '@/types';

interface HeadingProps {
  text: string;
  level?: 1 | 2 | 3;
  size?: TextSize;
  spanContent?: string;
}

const Heading = ({ level = 2, size = TextSize.Large, text, spanContent }: HeadingProps) => {
  const Tag = `h${level}` as keyof JSX.IntrinsicElements;

  const headingClassName = `text-${size} font-bold text-primary`;

  return (
    <>
      {level === 1 ? (
        <h1 className={headingClassName}>
          {spanContent && <span className="font-light">{spanContent}</span>} <br />
          {text}
        </h1>
      ) : (
        <Tag className={headingClassName}>{text}</Tag>
      )}
    </>
  );
};

export default Heading;
