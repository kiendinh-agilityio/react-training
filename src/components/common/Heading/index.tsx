interface HeadingProps {
  text: string;
}

const Heading = ({ text }: HeadingProps) => (
  <h1 className="flex justify-center font-helveticaBold text-dark text-[18px] leading-[25px] font-bold">
    {text}
  </h1>
);

export default Heading;
