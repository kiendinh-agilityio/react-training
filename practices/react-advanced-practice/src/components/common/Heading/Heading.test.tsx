import { render } from '@testing-library/react';

// Component Heading
import Heading from '.';

describe('Heading Component', () => {
  it('should match with snapshot', () => {
    const { container } = render(<Heading text="This is a heading" />);

    expect(container).toMatchSnapshot();
  });
});
