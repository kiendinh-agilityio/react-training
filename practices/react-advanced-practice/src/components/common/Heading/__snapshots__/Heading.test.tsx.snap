// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component should match with snapshot 1`] = `
<div>
  <h1
    class="rt-Heading rt-r-size-6 rt-r-weight-bold text-lg font-bold text-dark "
  >
    This is a heading
  </h1>
</div>
`;
